prow/cmd/hook/main.go

```diff
@@ -91,9 +91,12 @@ func main() {
 	}
 
 	pluginAgent := &plugins.PluginAgent{
-		GitHubClient: githubClient,
-		KubeClient:   kubeClient,
-		JobAgent:     jobAgent,
+		PluginClient: plugins.PluginClient{
```

Comment:
It felt funny passing a full JobAgent to the plugins. It might be worth pulling this out into a separate package, which has some other benefits too.